### DISCUSSION TITLE
Consider strictly following SemVer

### DISCUSSION THREAD

**Discussion by max-wittig** [python-poetry/poetry#4271] [General]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Issue

Might be a separate issue, but it seems poetry 1.0.10 is not working anymore for new projects. 
With this **minor** release from **1.0.10** to **1.1.X** lockfiles are actually not compatible anymore as the format was changed in `1.1`. According to SemVer such breaking changes would require a new **major** version (2.X).

In a patch release (`1.1.4`), the new installer was added and enabled by default, which I would feel like would at least be worth a minor release. We still have to disable the new installer in some of our projects, as it breaks (https://github.com/python-poetry/poetry/issues/3336)

Is it planned to follow SemVer more strictly with poetry? That would make it easier for everybody to upgrade faster.

Don't fear high version numbers.

@dlouzan @ercanucan

**Comment by finswimmer** [comment]
Hello @max-wittig,

poetry tries to follow SemVer. But as discussed in several places there will be always changes that lead to a disagreement between different people which version number would be correct.

Related to your points:

poetry 1.1.* can handle lock file created with poetry < 1.1 and >=1.1. So this a backwards compatibility and as such a minor version bump is correct.

Regarding the installer: Whether it should be a patch, minor or major  version is always related to the intention of maintainer. Even a small bugfix can be a breaking change to someone else (This is discussed in several places as well). The intention with the new installer wasn't to break someones work, but to change how something is done. (I'm still not able to reproduce the problems reported by others :thinking: )

> Don't fear high version numbers.

I fully agree here with you :smile: 

fin swimmer

PS: I will convert this issue to a discussion
